backend/src/charityprojects/api/views.py

```diff
@@ -576,14 +556,98 @@ def join_project_invitation(request):
         project_user.save()
         pu_id = project_user.id
         inviter_user_record = ProjectUser.objects.filter(project_id_id=project_id, user_id_id=inviter_user_id)[0].id
-        inviter_user_details = ProjectUserDetails.objects.get(pu_id=inviter_user_record)
-        prize_given_id = inviter_user_details.prize_given_id
+        prize_given_id = find_user_prize(inviter_user_record)
         project_user_details = ProjectUserDetails.objects.create(pu_id=pu_id, prize_given_id=prize_given_id)
         project_user_details.save()
         user_invitation = UserInvitation.objects.filter(pu_id_id=inviter_user_record)
         user_invitation.status = "Accepted"
         user_invitation.save()
-        response["Status"] = "Success"
+        response["status"] = "Success"
+    return JsonResponse(response)
+
+
+@api_view(['POST'])
+@parser_classes([MultiPartParser, FormParser])
+def spread_the_word(request):
+    response = {'status': "Invalid Request"}
+    project_id = request.data["project_id"]
+    user_email = request.data["user_email"]
+    message = request.data["invite_message"]
+    user_id = User.objects.get(email=user_email).id
+    project_user_record = ProjectUser.objects.filter(user_id_id=user_id, project_id_id=project_id)[0]
+    pu_id = project_user_record.id
+    prize_id = find_user_prize(pu_id)
+    project_user_details = ProjectUserDetails.objects.get(pu_id_id=pu_id)
+    if "video" in request.data:
+        project_user_details.video = request.data["video"]
+        project_user_details.save()
+
+    registered_user_list = request.data["registered_user"]
+    unregistered_user_list = request.data["unregistered_user"]
+
+    unregistered_user_invite = [item for item in unregistered_user_list if len(item) > 1 and item != user_email]
+    unregistered_user_invite = set(unregistered_user_invite)
+    for user_email in unregistered_user_invite:
+        invited_user = User.objects.get(email=user_email)
+        if invited_user:
+            registered_user_list.append(invited_user)
+        else:
+            create_unregister_user_invitation(user_email, pu_id, prize_id, message)
+
+    invited_users = [item for item in registered_user_list if len(item) > 1 and item != user_email]
+    invited_users = set(invited_users)
+    for user_email in invited_users:
+        if check_existing_project(user_email, project_id):
+            response["status"] = "User is already doing project"
+        else:
+            if create_user_invitation(user_email, pu_id, prize_id, message):
+                response["status"] = "Successfully stored invitation"
```

Comment:
This status can also be stored for each user, if it makes sense to store for each user. Nit sure though